lib/options.nix

```diff
@@ -45,158 +47,262 @@ let
+      `example`
+      : Optional example value used in the manual.
 
-     Example:
-       mkOption { }  // => { _type = "option"; }
-       mkOption { default = "foo"; } // => { _type = "option"; default = "foo"; }
+      `description`
+      : Optional string describing the option.
+
+      `relatedPackages`
+      : Optional related packages used in the manual (see `genRelatedPackages` in `../nixos/lib/make-options-doc/default.nix`).
+
+      `type`
+      : Optional option type, providing type-checking and value merging.
+
+      `apply`
+      : Optional function that converts the option value to something else.
+
+      `internal`
+      : Optional boolean indicating whether the option is for NixOS developers only.
+
+      `visible`
+      : Optional boolean indicating whether the option shows up in the manual. Default: true. Use false to hide the option and any sub-options from submodules. Use "shallow" to hide only sub-options.
+
+      `readOnly`
+      : Optional boolean indicating whether the option can be set only once.
+
+      `...` (any other attribute)
+      : Any other attribute is passed through to the resulting option attribute set.
+
+    # Examples
+    :::{.example}
+    ## `lib.options.mkOption` usage example
+
+    ```nix
+    mkOption { }  // => { _type = "option"; }
+    mkOption { default = "foo"; } // => { _type = "option"; default = "foo"; }
+    ```
+
+    :::
   */
   mkOption =
     {
-    # Default value used when no definition is given in the configuration.
-    default ? null,
-    # Textual representation of the default, for the manual.
-    defaultText ? null,
-    # Example value used in the manual.
-    example ? null,
-    # String describing the option.
-    description ? null,
-    # Related packages used in the manual (see `genRelatedPackages` in ../nixos/lib/make-options-doc/default.nix).
-    relatedPackages ? null,
-    # Option type, providing type-checking and value merging.
-    type ? null,
-    # Function that converts the option value to something else.
-    apply ? null,
-    # Whether the option is for NixOS developers only.
-    internal ? null,
-    # Whether the option shows up in the manual. Default: true. Use false to hide the option and any sub-options from submodules. Use "shallow" to hide only sub-options.
-    visible ? null,
-    # Whether the option can be set only once
-    readOnly ? null,
+      default ? null,
+      defaultText ? null,
+      example ? null,
+      description ? null,
+      relatedPackages ? null,
+      type ? null,
+      apply ? null,
+      internal ? null,
+      visible ? null,
+      readOnly ? null,
     } @ attrs:
     attrs // { _type = "option"; };
 
-  /* Creates an Option attribute set for a boolean value option i.e an
-     option to be toggled on or off:
+  /**
+    Creates an Option attribute set for a boolean value option i.e an
+    option to be toggled on or off.
+
+    default
+    : `false`
```

Comment:
This looks structured, but isn't.
These docs are supposed to be in a general Nix-level syntax, so we shouldn't be inventing module system-level doc syntax.

Reply:
Sorry i dont get it.

This is definition list syntax for commonmark. Which gets rendered nicely in the documentation.

Wouldn't a user want to know the default of `mkEnableOption` ? What is your proposed alternative here?